File: src/PHP/BehavioralPatterns/Mediator/ChainReaction/TextBox.php
<?php

namespace BehavioralPatterns\Mediator\ChainReaction;

class TextBox extends Participant {
    private bool $isEnabled = false;

    public function __construct(Mediator $mediator) {
        parent::__construct($mediator);
    }

    public function isEnabled(): bool
    {
        return $this->isEnabled;
    }

    public function enable(): void
    {
        $this->isEnabled = true;
        echo "TextBox is enabled." . PHP_EOL;
        $this->mediator->onStateChanged($this);
    }

    public function disable(): void
    {
        $this->isEnabled = false;
        echo "TextBox is disabled." . PHP_EOL;
        $this->mediator->onStateChanged($this);
    }
}

File: src/C#/BehavioralPatterns/Mediator/ChainReaction/TextBox.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ChainReaction
{
    public class TextBox : Participant
    {
        public TextBox(Mediator mediator) : base(mediator)
        {

        }

        public bool IsEnabled { get; set; } = false;

        public void Enable()
        {
            IsEnabled = true;
            Console.WriteLine("TextBox is enabled.");
            Mediator.OnStateChanged(this);
        }

        public void Disable()
        {
            IsEnabled = false;
            Console.WriteLine("TextBox is disabled.");
            Mediator.OnStateChanged(this);
        }
    }
}
